### ISSUE TITLE
Python 3 Support in pScheduler

### ISSUE THREAD

**Issue by arlake228** [perfsonar/pscheduler#483]
See title. This is the way the world seems to be going so we will need to do this sometime in the not-so-distant future. Putting in Icebox for now, but don't want to forget about it.

-----

Known TODOs:

- [x] Test Esmond archiver (Andy)
- [x] Test SNMP tests/tools/archiver (Ed)
- [x] Debian packaging (Antoine)

Known issues:

- [x] See why server isn't started after install (Mark)
- [x] See why scheduler runs slower (Mark)
- [x] Limit processor can't find library modules
- [x] See what happens when a tool enumerates a test that's not installed.  (`curl` being the poster child).

**Comment by vvidic** [comment]
Python 2.7 is supposed to be EOL 2020 and might not be included in the next Debian version. Python3 ships with a 2to3 tool to help with the conversion.

CentOS 7 does not ship python3 but epel contains python3.4. Supported libraries might also be a problem (only ~300 compared to 1500 for python2).

Default python 3 versions included in:
* Debian 8: python 3.4
* Debian 9: python 3.5
* Debian next: python 3.6
* Ubuntu 14: python 3.4
* Ubuntu 16: python 3.5
* Ubuntu next: python 3.6

**Comment by tonin** [comment]
I think we're still missing a few `python -m compileall` call replacements with the `$(PYTHON)` variable as we have in the `pscheduler-server/api-server/Makefile` I see that in most/all of the tests and tools and in the esmond-archiver.

**Comment by mfeit-internet2** [comment]
@tonin What you found has been fixed by cf8c1a5.

Change your build process so that any `make` doing `python -m compileall` passes `PYTHON=xxx` on the `make` command line as the RPM specs now do:

```
build
make \
     PYTHON=%{_pscheduler_python} \
     DESTDIR=$RPM_BUILD_ROOT/%{dest} \
     install
```
